### PR TITLE
Add GO Lang support

### PR DESCRIPTION
**Pull request by harshsharma2002**
- Add GO Lang support along with go lang tests.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,11 +10,22 @@ RUN set -ex && \
     rm -f /usr/libexec/gcc/x86_64-alpine-linux-musl/6.4.0/lto-wrapper && \
     rm -f /usr/bin/x86_64-alpine-linux-musl-gcj
 
+RUN set -ex && \
+    apk add --no-cache gcc g++ musl-dev python3 openjdk17 iptables ip6tables && \
+    apk add --no-cache --virtual .build-deps bash curl git && \
```
Comment (anipnwr7777) [inline Dockerfile:15]: Why do you want to add `bash`, `curl` and `git`?

File: Dockerfile [hunk 2]
```diff
@@ -10,11 +10,22 @@ RUN set -ex && \
...
+    rm go1.21.6.linux-amd64.tar.gz && \
+    apk del .build-deps
+
 RUN ln -sf python3 /usr/bin/python
 
 ADD . /usr/bin/
 ADD start.sh /usr/bin/
 
+ENV PATH=$PATH:/usr/local/go/bin
+ENV GOPATH=/go
```
Comment (anipnwr7777) [inline Dockerfile:27]: This is also not needed.

File: tests/data/testJson.js [hunk 3]
```diff
@@ -183,6 +183,47 @@ const testCases = [
...
+    {
+        name: 'golang : hello world',
+        reqObject: {
+            language: 'golang',
+            script:
+            'package main\n' +
+            'import "fmt"\n' +
+            'func main() {\n' +
+            'fmt.Println("hello world")\n' +
+            '}\n'
```
Comment (anipnwr7777) [inline tests/data/testJson.js:195]: add one tab of indentation.

File: tests/data/testJson.js [hunk 4]
```diff
@@ -183,6 +183,47 @@ const testCases = [
...
+            'import (\n' +
+            '"fmt"\n' +
+            '"math"\n' +
+            ')\n' +
+            'func main() {\n' +
+            'var number float64\n' +
+            'fmt.Scan(&number)\n' +
+            'result := math.Sqrt(number)\n' +
+            'fmt.Printf("%.2f",result)\n' +
+            '}\n',
```
Comment (anipnwr7777) [inline tests/data/testJson.js:218]: add one tab of indentation.

**Review by anipnwr7777 — CHANGES_REQUESTED** [review]
please address comments

File: Dockerfile [hunk 5]
```diff
@@ -10,11 +10,22 @@ RUN set -ex && \
     rm -f /usr/libexec/gcc/x86_64-alpine-linux-musl/6.4.0/lto-wrapper && \
     rm -f /usr/bin/x86_64-alpine-linux-musl-gcj
 
+RUN set -ex && \
+    apk add --no-cache gcc g++ musl-dev python3 openjdk17 iptables ip6tables && \
+    apk add --no-cache --virtual .build-deps bash curl git && \
+    curl -O https://dl.google.com/go/go1.21.6.linux-amd64.tar.gz && \
+    tar -C /usr/local -xzf go1.21.6.linux-amd64.tar.gz && \
+    rm go1.21.6.linux-amd64.tar.gz && \
+    apk del .build-deps
```
[the remark this answers is not in the document]
Comment (harshsharma2002) [reply Dockerfile:19]: that will work but it install's go lang version 1.17 to install the latest i had to curl as the latest available to alpine is 1.17. if 1.17 is fine i will revert.

File: Dockerfile (hunk 1 above)
Comment (harshsharma2002) [reply Dockerfile:15]: sorry, only curl will be needed to download the tar for latest go compiler.

File: Dockerfile (hunk 2 above)
Comment (harshsharma2002) [reply Dockerfile:27]: without 
`ENV PATH=$PATH:/usr/local/go/bin`
this line go lang compiler is not recognized by doker.

**Comment by harshsharma2002** [comment]
Please check the comments so i can make the required changes.

File: Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (anipnwr7777) [reply Dockerfile:19]: no it's fine.

**Comment by anipnwr7777** [comment]
we need `go` to work with the given configuration of `ALLOWED_RAM=512`, can you please look up for possible ways to restrict `go` to use less memory.

**Comment by anilgkalvium** [comment]
@anipnwr7777 would ulimit not work? with that in place is there an issue?

**Comment by harshsharma2002** [comment]
i am trying to make `GOMEMLIMIT` work but no luck so far.

**Comment by anipnwr7777** [comment]
> @anipnwr7777 would ulimit not work? with that in place is there an issue?

@anilgkalvium in-fact `ulimit` restriction of resources is creating the issue. We need a way to tell `go` to somehow use less resources and not restrict it using `ulimit`. if there is no possible way out then setting extra limits specially for `go` is the way out.

**Comment by anipnwr7777** [comment]
@harshsharma2002 I will be taking this forward. Thanks for your time and effort.

**Pull request closed without merging**